src/anim/anim.js

```diff
@@ -1,1132 +1,1096 @@
-Object.assign(pc, function () {
-
-    /**
-     * @private
-     * @constant
-     * @type {number}
-     * @name pc.INTERPOLATION_STEP
-     * @description A stepped interpolation scheme.
-     */
-    var INTERPOLATION_STEP = 0;
-
-    /**
-     * @private
-     * @constant
-     * @type {number}
-     * @name pc.INTERPOLATION_LINEAR
-     * @description A linear interpolation scheme.
-     */
-    var INTERPOLATION_LINEAR = 1;
-
-    /**
-     * @private
-     * @constant
-     * @type {number}
-     * @name pc.INTERPOLATION_CUBIC
-     * @description A cubic spline interpolation scheme.
-     */
-    var INTERPOLATION_CUBIC = 2;
-
-    /**
-     * @private
-     * @class
-     * @name pc.AnimData
-     * @classdesc Wraps a set of data used in animation.
-     * @description Create a new animation data container.
-     * @param {number} components - Specifies how many components make up an element of data.
-     * For example, specify 3 for a set of 3-dimensional vectors. The number of elements in
-     * data array must be a multiple of components.
-     * @param {Float32Array|number[]} data - The set of data
-     * @property {number} components - The number of components that make up and element
-     * @property {Float32Array|number[]} data - The data
-     */
-    var AnimData = function (components, data) {
-        this._components = components;
-        this._data = data;
-    };
-
-    Object.defineProperties(AnimData.prototype, {
-        components: {
-            get: function () {
-                return this._components;
-            }
-        },
-        data: {
-            get: function () {
-                return this._data;
-            }
+import { INTERPOLATION_CUBIC, INTERPOLATION_LINEAR, INTERPOLATION_STEP } from './constants.js';
+
+import { math } from '../math/math.js';
+
+import { AnimPropertyLocator } from '../framework/components/anim/property-locator.js';
+import { Entity } from '../framework/entity.js';
+
+/**
+ * @private
+ * @class
+ * @name pc.AnimData
+ * @classdesc Wraps a set of data used in animation.
+ * @description Create a new animation data container.
+ * @param {number} components - Specifies how many components make up an element of data.
+ * For example, specify 3 for a set of 3-dimensional vectors. The number of elements in
+ * data array must be a multiple of components.
+ * @param {Float32Array|number[]} data - The set of data
+ * @property {number} components - The number of components that make up and element
+ * @property {Float32Array|number[]} data - The data
+ */
+function AnimData(components, data) {
```

Comment:
Is this necessary for rollup to or a stylistic preference?

Reply:
It's not a rollup requirement. It's just more succinct (and IMHO, it's easier on the eyes). And currently, the engine uses both styles randomly. This PR makes all constructors consistent.